src/azure-cli-core/azure/cli/core/tests/test_parser.py

```diff
@@ -49,9 +49,9 @@ def test_handler2():
         args = parser.parse_args('sub-command the-second-name'.split())
         self.assertIs(args.func, command2)
 
-        AzCliCommandParser.error = VerifyError(self,)
-        parser.parse_args('sub-command'.split())
-        self.assertTrue(AzCliCommandParser.error.called)
+        with mock.patch('azure.cli.core.parser.AzCliCommandParser.error', side_effect=VerifyError(self)):
```

Comment:
Let's use [`new`](https://docs.python.org/3/library/unittest.mock.html#patch), otherwise `VerifyError` will be wrapped in a `MagicMock`, which is unnecessary. 

```suggestion
        with mock.patch('azure.cli.core.parser.AzCliCommandParser.error', new=VerifyError(self)):
```